From d73828faee24189e4fc5d326941fd2fbb236cff4 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Wed, 23 Mar 2022 15:35:10 +1200
Subject: [PATCH 10/17] cacheflush.h: Add forward declaration for struct folio

The struct folio is not declared in cacheflush.h so we need to
provide a forward declaration as otherwise users of this header
file may get warnings.

Reported-by: Guenter Roeck <linux@roeck-us.net>
Fixes: 522a0032af00 ("Add linux/cacheflush.h")
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Reviewed-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Link: https://lore.kernel.org/r/YjqVblTmjNYl3Zjc@gondor.apana.org.au
---
 include/linux/cacheflush.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/cacheflush.h b/include/linux/cacheflush.h
index fef8b607f97e..a6189d21f2ba 100644
--- a/include/linux/cacheflush.h
+++ b/include/linux/cacheflush.h
@@ -4,6 +4,8 @@
 
 #include <asm/cacheflush.h>
 
+struct folio;
+
 #if ARCH_IMPLEMENTS_FLUSH_DCACHE_PAGE
 #ifndef ARCH_IMPLEMENTS_FLUSH_DCACHE_FOLIO
 void flush_dcache_folio(struct folio *folio);
-- 
2.35.1

